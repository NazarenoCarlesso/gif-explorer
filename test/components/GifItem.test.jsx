import { render, screen } from '@testing-library/react'
import GifItem from '../../src/components/GifItem'

describe('Test <GifItem />', () => {
  const title = 'Dragon Ball Ultra Instinct GIF'
  const url = 'https://media2.giphy.com/media/19JSJ5ucu91R5D7a3w/giphy.gif?cid=b112ed82qn0pe3ma98puryu7iwn63w4eogbyoxq81m77kktt&rid=giphy.gif&ct=g'

  test('should match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)

    expect(container).toMatchSnapshot()
  })

  test('should have the url and alt', () => {
    render(<GifItem title={title} url={url} />)
    // screen.debug()
    // expect(screen.getByRole('img').src).toBe(url)
    // expect(screen.getByRole('img').alt).toBe(title)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('should have title as text', () => {
    render(<GifItem title={title} url={url} />)

    expect(screen.getByText(title)).toBeTruthy()
  })
})
