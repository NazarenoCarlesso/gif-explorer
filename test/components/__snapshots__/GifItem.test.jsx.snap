// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball Ultra Instinct GIF"
      src="https://media2.giphy.com/media/19JSJ5ucu91R5D7a3w/giphy.gif?cid=b112ed82qn0pe3ma98puryu7iwn63w4eogbyoxq81m77kktt&rid=giphy.gif&ct=g"
    />
    <h4>
      Dragon Ball Ultra Instinct GIF
    </h4>
  </div>
</div>
`;

exports[`Test <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball Ultra Instinct GIF"
      src="https://media2.giphy.com/media/19JSJ5ucu91R5D7a3w/giphy.gif?cid=b112ed82qn0pe3ma98puryu7iwn63w4eogbyoxq81m77kktt&rid=giphy.gif&ct=g"
    />
    <h4>
      Dragon Ball Ultra Instinct GIF
    </h4>
  </div>
</div>
`;
